pkg/api/v1beta1/conversion.go

```diff
@@ -444,7 +444,115 @@ func init() {
 			}
 			return nil
 		},
+		// Converts cpu and memory requirements to Resources, and vice versa.
+		func(in *newer.Container, out *Container, s conversion.Scope) error {
+			if err := s.Convert(&in.Name, &out.Name, 0); err != nil {
```

Comment:
+1. This is error prone and makes extending Containers struct tricky. I am all ears for a better approaches to handling this conversion logic. Happy to implement a better solution as well.